packages/dev/core/src/XR/features/WebXRAnchorSystem.ts

```diff
@@ -21,6 +21,12 @@ export interface IWebXRAnchorSystemOptions {
      * If not defined, anchors will be removed from the array when the feature is detached or the session ended.
      */
     doNotRemoveAnchorsOnSessionEnded?: boolean;
+
+    /**
+     * Not setting this flag will clear the anchors array when the session is initialized.
+     * Note that the native anchors will not be removed. The onl thing cleared would be the anchors array of the feature itself.
+     */
+    doNotClearAnchorsOnSessionInit?: boolean;

```

Comment:
Why do you need a new option for this and why does it need to be done in init? Does this result in different behavior than just clearing without deleting on session ended?

Reply:
The idea was to try not to break the behavior as much as possible. The default behavior was to remove all anchors on session end, which means that those arrays will be cleared. Now that the delete on session end doesn't happen, I wanted to be sure that when the session starts, the feature is "reset" and doesn't have the old anchors' references.

Reply:
I changed the entire behavior. I kept the new flag, reversed, just in case anyone finds it useful, but this is an opt-in flag.
To fix the issue - I simply avoid calling the remove function of the anchor, as it anyhow makes no sense doing it - it is already removed from the session.